### ISSUE TITLE
missing atlas when I cmake .. -DUSE_MPI=ON

### ISSUE THREAD

**Issue by xyyu-callen** [yjxiong/caffe#109]
Hello @yjxiong. When I cmake .. -DUSE_MPI=ON. I encounter this error.
centos 7, CUDA 7.5 is my environment, And I have installed atlas-devel.
The following is my errors.

-- Added CUDA NVCC flags for: sm_35
-- OpenCV found (/usr/local/share/OpenCV)
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find Atlas (missing: Atlas_CBLAS_LIBRARY Atlas_BLAS_LIBRARY
  Atlas_LAPACK_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindAtlas.cmake:43 (find_package_handle_standard_args)
  cmake/Dependencies.cmake:74 (find_package)
  CMakeLists.txt:30 (include)

-- Configuring incomplete, errors occurred!

What should I do for solving this problem?
Best Regards

**Comment by yjxiong** [comment]
I think you need [this](https://github.com/BVLC/caffe/pull/2394) commit to make CMake to recognize it automatically. Or you can install `blas-devel` and `lapack-devel` to avoid modifying codebase.

**Comment by xyyu-callen** [comment]
Thank you very much. 
This problem has solved. 
But I have new one.

-- Found NumPy: /usr/lib64/python2.7/site-packages/numpy/core/include (found suitable version "1.7.1", minimum required is "1.7.1") 
-- NumPy ver. 1.7.1 found (include: /usr/lib64/python2.7/site-packages/numpy/core/include)
-- Could NOT find Boost
-- Found Doxygen: /bin/doxygen (found version "1.8.5") 
-- Could NOT find MPI_C (missing:  MPI_C_LIBRARIES MPI_C_INCLUDE_PATH) 
-- Could NOT find MPI_CXX (missing:  MPI_CXX_LIBRARIES MPI_CXX_INCLUDE_PATH) 
-- Python interface is disabled or not all required dependecies found. Building without it...
-- Detected Doxygen OUTPUT_DIRECTORY: ./doxygen/

## -- Found Git: /bin/git (found version "1.8.3.1") 

-- ******************\* Caffe Configuration Summary *******************
-- General:
--   Version           :   <TODO> (Caffe doesn't declare its version in headers)
--   Git               :   v0.9999-1566-g0c69f8b-dirty
--   System            :   Linux
--   C++ compiler      :   /bin/c++
--   Release CXX flags :   -O3 -DNDEBUG -fPIC -Wall -Wno-sign-compare -Wno-uninitialized
--   Debug CXX flags   :   -g -fPIC -Wall -Wno-sign-compare -Wno-uninitialized

## --   Build type        :   Release

--   BUILD_SHARED_LIBS :   ON
--   BUILD_python      :   ON
--   BUILD_matlab      :   OFF
--   BUILD_docs        :   ON

## --   CPU_ONLY          :   OFF

The above is my errors.
How do I fix this errors?
I am anticipating your reply.
Best regards.

**Comment by yjxiong** [comment]
What's the error?

**Comment by xyyu-callen** [comment]
Could NOT find Boost
Could NOT find MPI_C (missing: MPI_C_LIBRARIES MPI_C_INCLUDE_PATH) 
Could NOT find MPI_CXX (missing: MPI_CXX_LIBRARIES MPI_CXX_INCLUDE_PATH) 

what reason is caused?

**Comment by yjxiong** [comment]
Boost version 1.55+ is needed.
For mpi, see the [FAQ](https://github.com/yjxiong/caffe/wiki/Compilation).

**Comment by 2502572025** [comment]
Thank you very very much.
@yjxiong 
I have solved this problem.
Best Regards.

**Comment by yjxiong** [comment]
Closing this. Please feel free to reopen it if you meet any further problem.

**Comment by xyyu-callen** [comment]
@yjxiong 
Thank you very much.

For this problem:
Could NOT find MPI_C (missing: MPI_C_LIBRARIES MPI_C_INCLUDE_PATH) 
Could NOT find MPI_CXX (missing: MPI_CXX_LIBRARIES MPI_CXX_INCLUDE_PATH)
Solving:
cmake .. -DUSE_MPI=ON -DMPI_CXX_COMPILER='/usr/lib64/openmpi/bin/mpicxx' -DMPI_C_COMPILER='/usr/lib64/openmpi/bin/mpicc'

For this error:
Could NOT find Boost
Solving:
compile and install boost
add following 2 lines into caffe/cmake/Dependencies.camke before find_package(Boost)
SET(Boost_INCLUDE_DIR "/root/Downloads/boost_1_56_0")
SET(Boost_LIBRARIES "/root/Downloads/boost_1_56_0/lib")

Best Regards

**Comment by yjxiong** [comment]
Awesome. Thanks for posting the solution.

**Comment by 2502572025** [comment]
@yjxiong 
Sorry, I have new problem.
When I running, I get this error.

I0929 16:41:55.571084 11416 net.cpp:282] This network produces output accuracy
I0929 16:41:55.571087 11416 net.cpp:282] This network produces output loss
I0929 16:41:55.571123 11416 net.cpp:531] Collecting Learning Rate and Weight Decay.
I0929 16:41:55.571146 11416 net.cpp:294] Network initialization done.
I0929 16:41:55.571153 11416 net.cpp:295] Memory required for data: 594577388
I0929 16:41:55.571342 11416 solver.cpp:47] Solver scaffolding done.
I0929 16:41:55.571410 11416 solver.cpp:363] Solving CUHK_Action_TemporalNet
I0929 16:41:55.571418 11416 solver.cpp:364] Learning Rate Policy: step
I0929 16:41:56.186375 11416 solver.cpp:335] Model Synchronization Communication time 0.614916 second
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 11417 on node localhost exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------


What reason?
What do I do?
